tools/AutoTuner/README.md

```diff
@@ -0,0 +1,205 @@
+gcloud beta container node-pools create --help
+```
+
+```bash
+gcloud beta container clusters create "autotuner" \
+         --machine-type "e2-standard-8" \
+         --image-type "UBUNTU_CONTAINERD" \
+         --disk-type "pd-standard" \
+         --disk-size "100" \
+         --num-nodes "1" \
+         --enable-autoscaling \
+         --min-nodes "1" \
+         --max-nodes "10"
+```
+
+Configure `gcloud` to use the Kubernetes credentials of the newly created
+cluster. The credentials allow the use of `kubectl` locally.
+
+```bash
+gcloud components install kubectl
+gcloud container clusters get-credentials autotuner
+```
+
+Create a new node pool. This step is optional, but recommended. The new pool
+will use preemptive nodes which are cheaper, but also has a more powerful
+CPU and have more disk space.
+
+```bash
+gcloud beta container node-pools create "worker-pool" \
+         --cluster "autotuner" \
+         --machine-type "c2-standard-60" \
+         --image-type "UBUNTU_CONTAINERD" \
+         --disk-type "pd-standard" \
+         --disk-size "2000" \
+         --preemptible \
+         --num-nodes "3" \
+         --enable-autoscaling \
+         --min-nodes "1" \
+         --max-nodes "625"
+```
+
+## Create NFS and setup mount options
+
+NOTE: This tutorial requires a working NFS server.
+
+To access the NFS mount point in the Ray cluster we use a NFS helm chart. You
+need to modify the helm chart with the information about your server IP
+and mount point path.
+
+### Install NFS Kubernetes provisioner
+
+```bash
+helm repo add nfs-subdir-external-provisioner https://kubernetes-sigs.github.io/nfs-subdir-external-provisioner
+helm repo update
+```
+
+Before installing the helm chart, double check the fields marked with
+`TODO` on the file `./nfs/nfs-helm-values.yaml`. The `server` and `path`
+need to match the NFS server IP and the path to the exposed folder.
+The `nodeSelector` must match the non-preemptive Kubernetes pool.
+
+```bash
+helm install nfs nfs-subdir-external-provisioner/nfs-subdir-external-provisioner -f nfs/nfs-helm-values.yaml
+```
+
+### Create a PVC
+
+Create the PVC:
+
+```bash
+kubectl create -f nfs/nfs-pvc.yaml
+```
+
+## Ray cluster
+
+The two main files for the helm chart are:
+
+```bash
+helm-chart/templates/raycluster.yaml
+helm-chart/values.yaml
+```
+
+To restrict to a specific node pool, replace POOL_NAME with the name
+of the pool you created on the previous step. For this tutorial we use
+`worker-pool`.
+
+```yaml
+nodeSelector:
+  cloud.google.com/gke-nodepool: POOL_NAME
+```
+
+### Deploy Ray cluster inside Kubernetes
+
+Use Helm to deploy a Ray cluster.
+
+```bash
+helm install autotuner ./helm-chart
+```
+
+### Upgrade deployment
+
+After the initial deployment, if you change the values inside `./helm-chart`
+you will need to upgrade the configuration stored at the Kubernetes cluster.
+
+```bash
+helm upgrade autotuner ./helm-chart
+```
+
+### Useful ways to observe
```

Comment:
Useful ways to dynamically view process run results and Status (just more specific) ..feel free to modify if long